cli/src/lib.rs

```diff
@@ -938,7 +949,7 @@ fn new(
 
                 let cluster = cfg.provider.cluster.clone();
                 let programs = cfg.programs.entry(cluster).or_default();
-                if programs.contains_key(&name) {
+                if !force && programs.contains_key(&name) {
```

Comment:
Thanks for updating but I don't think this would work as expected because we don't remove any of the existing files(with both commands). Did you test this?

Reply:
Good point. I interpreted the issue as not being able to force create if a folder already exists and not necessarily remove all the files within. Does that make sense? Under this scenario this actually works.

However, I'm adjusting the `init` command to actually skip the folder creation piece.

Reply:
Done at https://github.com/coral-xyz/anchor/pull/2698/commits/ba6b7c90e46f868c1d6d2b5a150d3d0294e11a8c

Reply:
>  I interpreted the issue as not being able to force create if a folder already exists

We are not force creating it though, it will still fail because [`std::fs::create_dir`](https://doc.rust-lang.org/std/fs/fn.create_dir.html) returns an error if the path already exists.

If I have program named `example`, why would I run `anchor new example --force`? To me, the answer is because I'd like to override the existing `example` program and create a new one. Maybe I'm missing something here but why else would I run it with `--force`?

```
$ anchor new example --force
Created new program.
```

but it didn't create a new program, pretty much nothing changed.

Reply:
Got it. My understanding and use case is: I already have a folder and other files within that folder that I want to now add a workspace or program to. Right now without the `force` that's not possible. With these changes that becomes possible.

What you're suggesting is that within that folder there's already a solana workspace or program and everything should be removed prior to initialization. Correct? I'm happy to make the change to accommodate what you mentioned.

Maybe the ideal scenario is to actually keep the files that do not conflict with the `init` or `new` process and overwrite all others. What do you think?

Reply:
> I already have a folder and other files within that folder that I want to now add a workspace or program to.

Thanks, that also makes sense.

> What you're suggesting is that within that folder there's already an anchor workspace or program and everything should be removed prior to initialization. Correct?

Yes, I think using `--force` is a bit unintuitive here both for `init` and `new` commands because for `init`, it errors due to existing items and for `new`, nothing happens.

It also doesn't have to be a full workspace because a single empty directory(e.g. `tests`) can make `anchor init` fail.

> Maybe the ideal scenario is to actually keep the files that do not conflict with the init or new process and overwrite all others. What do you think?

The confusing part(at least for me) is the `force` flag name because it's usually used when you really want to force something to happen — like in the previous example I'd expect `anchor new example --force` to create a new program named `example` regardless of whether it exists or not.

Reply:
Completely agree. Do you mind checking the last commits?

It may be a good compromise between the options above:
- For `new --force` it will delete the new program folder that it's trying to create and create from scratch
- For `init --force` it will delete the default program as well and for all other folders and files it will just rewrite them if they already exist leaving all other files untouched

Reply:
I think this is more intuitive now, thank you for updating.